### PR TITLE
[MSN-335] Airborne DSA Revival

### PR DESCRIPTION
**Pull request by tomeldar**
## Ticket

[MSN-335](https://intterragroup.atlassian.net/browse/MSN-335)

## Background

This is setting up a CLI intended for use by operators on a plane while collecting imagery, videos, and tactical products for a live mission. It will create the scaffolding for the mission based on the name and time and then watch for new files to upload to S3 accordingly.

### Points of Focus

- Github actions. Unit tests are running accordingly and I've verified a now deleted release tag generated the appropriate windows executable
- We have a product model to represent airborne products
- Core logic is in the main.py file. It is supplemented by services in the service module

### Test Plan

- Jim will be the primary tester on this
- If you're a developer running this locally, set "storageMode": "local" in config.json. You can then use a fake bucket name and fake AWS credentials to test

## Optional Fun

I'm somewhat of a collector myself

[MSN-335]: https://intterragroup.atlassian.net/browse/MSN-335?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by SageVanGogt** [comment]
Looking forward to taking this for a little test ride tomorrow. I did give the code a review, and it looks good, though I don't know python so I can't really critique that. I don't know if this is being used for a specific client or what but we should probably add support for all the tactical product types that we support in the airborne etl.

**Comment by tomeldar** [comment]
> Looking forward to taking this for a little test ride tomorrow. I did give the code a review, and it looks good, though I don't know python so I can't really critique that. I don't know if this is being used for a specific client or what but we should probably add support for all the tactical product types that we support in the airborne etl.

Established vendors already have their own process for this, so it's less of a necessity for them. Main focus is creating a smooth onboarding experience for new clients. Good call on the tactical products, must have missed a couple

File: README.md [hunk 1]
```diff
@@ -1,14 +1,29 @@
...
 
 ## Getting started
-* Make sure you have python 3 installed
-* `pip install awscli boto3`
-* `aws configure`
-* Set the correct bucket name in `settings.py`
 
-Run: `python main.py` or `intterra-airborne-dsa.bat`
+- `source env/bin/activate`
+- `python3 -m python install -e .`
```
Comment (nronnei) [inline README.md]: Should this be `pip` instead of `python`?

Comment (tomeldar) [reply README.md]: Yes, will resolve

File: main.py [hunk 2]
```diff
@@ -1,146 +1,219 @@
...
+        config.aws_access_key_id, config.aws_secret_access_key, config.bucket
+    )
+
+    mission_name, mission_time = get_mission_details()
+    # Create mission file
+    try:
+        s3_client.upload_empty_file(
+            f"MISSION/{mission_name}_{mission_time.strftime('%Y%m%d_%H%M')}Z.txt"
+        )
+        print(f"Created mission: {mission_name}")
```
Comment (nronnei) [inline main.py]: I've been bitten several times by "comment out to test" before. I'd prefer to see a flag set via config. Here's a scrappy example of what I mean:

[storage-mode-flag.diff.txt](https://github.com/intterra-io/intterra-airborne-dsa/files/12397902/storage-mode-flag.diff.txt)

**Review by nronnei — APPROVED** [review]
My concerns have been addressed. Thanks for that! Looking good to me from a technical standpoint.

**Pull request merged**